index.js

```diff
@@ -31,22 +32,40 @@ function nestedProp(obj, path) {
 	return ret;
 }
 
-function inject(string, obj, regexp) {
-	return string.replace(regexp, (match, path) => nestedProp(obj, path));
+function inject(string, obj, re) {
+	return string.replace(re, (match, path) => nestedProperty(obj, path));
 }
 
-module.exports = postcss.plugin(PLUGIN_NAME, opts => {
+const externalVars = postcss.plugin(PLUGIN_NAME, opts => {
 	opts = opts || {};
 
-	const regexp = generateRegExp(opts.prefix || '$');
+	const re = makeRe(opts.prefix || '$');
 
 	return css => {
 		css.walkDecls(decl => {
 			try {
-				decl.value = inject(decl.value, opts.data, regexp);
+				decl.value = inject(decl.value, opts.data, re);
 			} catch (err) {
 				throw decl.error(err.message, {plugin: PLUGIN_NAME});
 			}
 		});
 	};
 });
+
+externalVars.test = (str, opts) => {
+	opts = opts || {};
+	const re = makeRe(opts.prefix || '$');
+
+	return re.test(str);
+};
```

Comment:
It might be worth considering memoising `makeRe` so as to not incur the performance overhead of constructing a new RegExp each time; however that comes with it's own complications (namely that the `lastIndex` is retained between matches), so if you did so you'd also want to switch from `re.test(str)` to `!!str.match(re)` or similar (it looks like `str.search(re) > -1` should be more performant than `!!str.match(re)`, but I'm not 100% certain it also doesn't have the `lastIndex` feature that `test`/`exec` have).